presto-main/src/test/java/io/prestosql/sql/planner/assertions/JoinMatcher.java

```diff
@@ -155,11 +156,15 @@ private boolean matchDynamicFilters(JoinNode joinNode, SymbolAliases symbolAlias
             if (build == null) {
                 return false;
             }
-            actual.add(new ComparisonExpression(descriptor.getOperator(), probe.toSymbolReference(), build.toSymbolReference()));
+            Expression expression = new ComparisonExpression(descriptor.getOperator(), probe.toSymbolReference(), build.toSymbolReference());
+            if (descriptor.isNullAllowed()) {
+                expression = new NotExpression(expression);
```

Comment:
Could we translate that into `new NotExpression(new CompaisonExpression(IS_DISTINCT_FROM, ...)`.
Translating it into `new NotExpression(new ComparisonExpression(EQUAL), ` as it is now is hacky